c10/core/DispatchKey.h

```diff
@@ -325,6 +325,7 @@ enum class DispatchKey : uint16_t {
   // Autocasting precedes VariableTypeId, to ensure casts are autograd-exposed
   // and inputs are saved for backward in the post-autocast type.
   AutocastCPU,
+  AutocastXPU,
```

Comment:
One thing I'm wondering - does XPU really need it's own set of custom autocast kernels? (so the rules are totally different vs the existing cpu / cuda rules?)

@albanD do you know who would be a good person to review extensions to AMP?

Reply:
@bdhirsh  Thanks for comments.
We will have similar solution for autocast, comparing with CPU and CUDA.
But the detail implementation is not identical.
1. Since CPU/CUDA is almost always available in PyTorch release package, perhaps we cannot reuse or override their dispatch key.
2. The backend capacity is different between AutocastXPU and other device. XPU will have different kernel list, comparing with CPU and CUDA.
3. XPU is out-of-tree device. Each XPU implementation will provide different kernel list, which depends on separate AutocastXPU key.